.github/workflows/test.yml

```diff
@@ -59,6 +61,44 @@ jobs:
         with:
           short-length: 7
 
+      - name: Authenticate to Google Cloud
+        id: auth
+        uses: google-github-actions/auth@v0.7.0
+        with:
+          workload_identity_provider: 'projects/143793276228/locations/global/workloadIdentityPools/github-actions/providers/github-oidc'
+          service_account: 'github-service-account@zealous-zebra.iam.gserviceaccount.com'
+          token_format: 'access_token'
+
+      # If we have accidentally changed the format, but not changed the version,
+      # we want to run with the old cached state, so other jobs fails.
+      #
+      # If we change the state path without changing the version,
+      # the regenerate-stateful-disks job will take a few hours because it will do a full rebuild.
+      - name: Get specific changed files
+        id: changed-files-specific
+        uses: tj-actions/changed-files@v18.6
+        with:
+          files: |
+            zebra-state/**/constants.rs
+
+      # Before executing any further steps, validate the local state and remote version are the same,
+      # or at least that the local state version is lower than the available cached state version.
+      #
+      # This step should only be validated if there are no changes in the files specifified
+      # in the changed-files-specific step.
+      - name: Validate constants.rs version vs cached state version
+        id: validate-state-version
+        if: ${{ steps.changed-files-specific.outputs.any_changed != 'true' }}
```

Comment:
Question: I'm not sure I understand why we are skipping this check in this test. What would happen if we always did the check?